### ISSUE TITLE
Fixed loss calculation in triplet mode to prevent potential model divergence.

### ISSUE THREAD

**Issue by VecherVhatuX** [VecherVhatuX/learning_data_science_in_prison#1075]
Fixed loss calculation in triplet mode to ensure non-negative result and prevent potential model divergence, changed loss calculation to use jnp.maximum to replace negative values with 0, ensuring stable and effective model training.
